### ISSUE TITLE
Loads on InputObject is not working with mutations 

### ISSUE THREAD

**Issue by michal-samluk** [rmosolgo/graphql-ruby#2280]
New version of graphql introduced load on input object. I think they might not be working properly for mutations.

My schema looks like this:
```ruby
class MyMutation < BaseMutation
  argument :parent_id, ID, required: true, loads: Types::ParentType
  argument :my_object, [Types::Inputs::MyInput], required: true
end

module Types
  module Inputs
    class MyInput < BaseInputObject
      argument(
        :child_id,
        ID,
        loads: Types::ChildType,
        required: true
      )
    end
  end
end

```
I am getting some weird behaviour for this. So my ParentType is loaded properly.
Then `resolve` method in mutation is called. Argument `my_object` looks like this:
```
[{
  child: "EncodedIdHere"
}]
``` 
So the key name was transformed (no _id suffix), but ID was not decoded. After the resolution `GraphqlSchema.object_from_id` is called with first argument being `nil`.

**Comment by cmschuetz** [comment]
I'm seeing these exact symptoms as well - mutations, `_id` is dropped correctly, value is encoded id instead of loaded object.

**Comment by rmosolgo** [comment]
Thanks for reporting this. Turns out the bug is that `loads:` is only applied when you call a _method_ on the input object. For exmaple: 

```diff 
  def resolve(my_object:)
- my_object[:child] # returns a string 
+ my_object.child # returns the loaded object 
```

You can make that change as a workaround, I'm working on a fix here: #2323
